explore/hash.h

```diff
@@ -112,14 +112,20 @@ inline uint64_t uniform_hash(const void * key, size_t len, uint64_t seed)
 
   switch (len & 3)
   {
-  case 3: k1 ^= tail[2] << 16;
-  case 2: k1 ^= tail[1] << 8;
+  case 3:
+    k1 ^= tail[2] << 16;
+    // fall through
```

Comment:
Does the presence of the comment suppress the warning, or does the command have to be exactly `// fall through`? If not the latter, could you add a note about this suppressing the warning, so it does not get removed later?

Reply:
Yep! GCC see's this as an annotation and silences the warning. There is also a gnu specific attribute available. In C++17 a [[fallthrough]] attribute was added for this exact purpose. 

I figured a comment is simple and suffices for now. I'll add a description so we don't accidentally remove it.